restful_api/myapi/api/resources/course.py

```diff
@@ -82,48 +83,48 @@ def post(self, course_id):
         return {"msg": "course created", "course": schema.dump(course)}, 201
 
 
-# -------
-# TODO: This is a work in progress, blocked till further notice
-# -------
 class CourseList(Resource):
-    """Get all courses
+    """Get all courses, with boolean indicating if engineer is eligible / enrolled
 
     ---
     get:
       tags:
         - api
+      parameters:
+        - name: eng_id
+          in: query
+          type: integer
+          required: true
+          description: engineer id
       responses:
+        400:
+          description: no engineer id provided
         200:
           content:
             application/json:
               schema:
-                allOf:
-                  - $ref: '#/components/schemas/PaginatedResult'
-                  - type: object
-                    properties:
-                      results:
-                        type: array
-                        items:
-                          $ref: '#/components/schemas/UserSchema'
+                type: object
+                properties:
+                  msg:
+                    type: string
+                    example: all courses retrieved
+                  result: CourseSchema
+            
     """
 
     method_decorators = [jwt_required()]
 
-    def get(self):
-      
-      # Step 1: Get all the courses
-      query = Course.query.all()
-      schema = CourseSchema(many=True)
-      courses = schema.dumps(query).data
-      
-      # Step 2: Get engineer completed courses
+    def get(self, eng_id):
       
-      # Step 3: Get enrolled courses
+      all_courses = Course.query.all()
+    
+      enrolled_courses = OfficialEnroll.query.filter(
+                   OfficialEnroll.eng_id==eng_id,
+                   ).all()
       
-      # Step 4: Validate the courses pre-req
+      course_schema = CourseSchema(many=True)
+      official_enroll_schema = OfficialEnrollSchema(many=True)
       
-      # Step 5: Convert back to MA format and paginate
+      courses = validate_prereqs(all_courses, official_enroll_schema.dump(enrolled_courses))
       
-      result = validate_prereqs(courses, completed_courses)
-      query = schema.load(result)
-      return paginate(query, schema)
+      return {"msg": "all courses retrieved", "results": course_schema.dump(courses)}, 200
```

Comment:
Note: I can't use paginate on this return, cos I need to preprocess `all_courses`, and paginate requires the input in query form.